src/main/java/com/employee/dao/EmployeeDAOImpl.java

```diff
@@ -164,4 +254,117 @@ public Employee getEmployeeById(int id) {
 		return employee;
 	}
 
+	public int getNumberOfTotalEmployee() {
+		int total = 0;
+		try (Connection conn = DBConnection.getConnection();
+				PreparedStatement preparedStatement = conn.prepareStatement(sqlSelectCount)) {
+			ResultSet resultSet = preparedStatement.executeQuery();
+			if (resultSet.next()) {
+				total = resultSet.getInt(1);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return total;
+	}
+
+	public String getAllEmployeeAvgHireDate() {
+		String avgHireDate = "";
+		try (Connection conn = DBConnection.getConnection();
+				PreparedStatement preparedStatement = conn.prepareStatement(sqlCalculateAvgHiredate)) {
+			ResultSet resultSet = preparedStatement.executeQuery();
+			if (resultSet.next()) {
+				avgHireDate = resultSet.getString(1);
+			}
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+		return avgHireDate;
+	}
+
+	public void updateEmployeesSalaryByBatch(Object[][] employeesToUpdate) {
+		try (Connection conn = DBConnection.getConnection();
+				PreparedStatement preparedStatement = conn.prepareStatement(sqlUpdateEmployeeSalary)) {
+
+			for (Object[] employee : employeesToUpdate) {
+				int id = (int) employee[0];
+				int salary = (int) employee[1];
+
+				preparedStatement.setLong(1, salary);
+				preparedStatement.setInt(2, id);
+
+				preparedStatement.addBatch();
+			}
+			preparedStatement.executeBatch();
+
+		} catch (SQLException e) {
+			e.printStackTrace();
+		}
+	}
+
+	public Boolean employeeBackup() {
+		Boolean result = false;
+		String backupFilePath = "D:/employee_backup.txt";
+		try (Connection conn = DBConnection.getConnection();
+				PreparedStatement preparedStatement = conn.prepareStatement(sqlSelectAll);
+				BufferedWriter writer = new BufferedWriter(new FileWriter(backupFilePath))) {
+
+			ResultSet resultSet = preparedStatement.executeQuery();
+
+			while (resultSet.next()) {
+				int employeeId = resultSet.getInt("employee_id");
+				String employeefName = resultSet.getString("first_name");
+				String employeelName = resultSet.getString("last_name");
+				String email = resultSet.getString("email");
+				String hireDate = resultSet.getString("hire_date");
+				double salary = resultSet.getDouble("salary");
+
+				String backupData = String.format("%d,%s,%s,%s,%s,%.2f%n", employeeId, employeefName, employeelName,
+						email, hireDate, salary);
+				writer.write(backupData);
+
+			}
+			result = true;
+
+		} catch (SQLException | IOException e) {
+			e.printStackTrace();
+			result = false;
+		}
+		return result;
+	}
+
+	public Boolean restoreEmployee(String backupFilePath) {
+		Boolean result = false;
+		try (Connection conn = DBConnection.getConnection();
```

Comment:
@hninpwintaung-hpa Connection object is unnecessary as the DB operation is done by other method.